Architecture n-tier

vlad12345123 opened the pull request:
Architecture n-tier implemented and added work with files

vlad12345123 committed:
remove last value

remove last value from summaries array

vlad12345123 committed:
Implementing project architecture

Implementing project architecture and basics things whats we need: EF core, Controllers, some patterns. Also the implementation of working with files

vlad12345123 committed:
Add ExaLearnDbContext

Remove appDbContext and add ExaLearnDbContext

vlad12345123 committed:
Merge branch 'VladZezjulin/dev' into vzezjulin/storemedia

vlad12345123 committed:
Implemented initial layered architecture (n-tier)

Architecture n-tier implemented and added work with files

romanstat commented on InternshipBE/ExaLearn.Bl/DTO/FileEntryDTO.cs:
```diff
@@ -0,0 +1,15 @@
+{
+    public class FileEntryDTO
+    {
+        //public int Id { get; set; }
```
remove comments and leave the fields

romanstat commented on InternshipBE/ExaLearn.Bl/Extensions/AudioExtensions.cs:
```diff
@@ -0,0 +1,16 @@
+{
+    internal static class AudioExtensions
+    {
+        public static readonly string[] Extensions =
```
rename

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,68 @@
+        public async Task<FileEntry> AddAsync(IFormFile file)
+        {
+            /*
+             do not forget to clarify the point regarding the naming of files!
```
comment

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,68 @@
+
+            var fileEntry = new FileEntry();
+
+            foreach (var format in AudioExtensions.Extensions)
```
Contains Linq

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,68 @@
+
+            foreach (var format in AudioExtensions.Extensions)
+            {
+                if (fileType.ToLower() == format) // realize this in enum
```
StringComparer

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,68 @@
+            {
+                if (fileType.ToLower() == format) // realize this in enum
+                {
+                    var filePath = "/Files/Audio/" + file.FileName; //absolute path to audiofile
```
const

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,68 @@
+                {
+                    var filePath = "/Files/Audio/" + file.FileName; //absolute path to audiofile
+
+                    if (file != null && file.Length >= 0)
```
upper

RomanMRC replied:
String.IsNullOrEmpty

romanstat commented on InternshipBE/ExaLearn.Dal/Database/ExaLearnDbContext.cs:
```diff
@@ -0,0 +1,15 @@
+{
+    public class ExaLearnDbContext : DbContext
+    {
+        public DbSet<FileEntry> FileEntry { get; set; }
```
bottom

romanstat commented on InternshipBE/ExaLearn.Dal/DbInitializer/DbInitialize.cs:
```diff
@@ -0,0 +1,17 @@
+    {
+        public static void Initialize(ExaLearnDbContext exaLearnDbContext)
+        {
+            exaLearnDbContext.Database.EnsureCreated(); // its create db before deploy our project on the server
```
remove

romanstat commented on InternshipBE/ExaLearn.Dal/Interfaces/IFileRepository.cs:
```diff
@@ -0,0 +1,15 @@
+{
+    public interface IFileRepository
+    {
+        Task<List<FileEntry>> GetAllFilesAsync();
```
spaces

romanstat commented on InternshipBE/ExaLearn.Dal/Model/FileEntry.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public class FileEntry
+    {
+        public int Id { get; set; }
```
spaces

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/FileRepository.cs:
```diff
@@ -0,0 +1,34 @@
+{
+    public class FileRepository : IFileRepository
+    {
+        private ExaLearnDbContext appDbContext;
```
space

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/FileRepository.cs:
```diff
@@ -0,0 +1,34 @@
+    public class FileRepository : IFileRepository
+    {
+        private ExaLearnDbContext appDbContext;
+        public FileRepository(ExaLearnDbContext appDbContext)
```
generic repository

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/Repository.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace ExaLearn.Dal.Repositories
+{
+    class Repository
```
implement generic repository

romanstat commented on InternshipBE/WebApi/Startup.cs:
```diff
@@ -1,16 +1,14 @@
-using System.Linq;
-using System.Threading.Tasks;
 
 namespace WebApi
```
ExaLearn.WebApi

ExadelSandbox commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,40 @@
+
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(int id)
```
Please, Avoid using the meaningless "Get" Does it mean Get files for listening or get file for estimation?

ExadelSandbox commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace WebApi.Controllers
+{
+    [Route("[controller]")]
```
it had better to remove the Route attribute from here and provide routing for each methods

vlad12345123 committed:
Project Edits

The following edits have been made: changed the repository to a generic repository, edits to the code convention, edits to the file logic

vlad12345123 committed:
editing missed changes

missed comment in DTO

vlad12345123 commented:
I do changes!

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+    public class FileService : IFileService
+    {
+        IGenericRepository<FileEntry> _filesRepository;
+        IConfiguration _configuration; //it should be in startup.. but for now i do it like this
```
space between ctor and field

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+            if (file != null && file.Length >= 0) 
+            {
+                var fileType = Path.GetExtension(file.FileName);
+                var isNessesaryFormat = AudioExtensions.AvalableAudutionExtensions.Any(e => e.Contains(fileType.ToLower()));
```
remove Any

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+                {
+                    var filePath = _configuration.GetConnectionString("HostingContext");
+
+                    using (var stream = new FileStream(Path.Combine(filePath, Path.GetRandomFileName()), FileMode.Create)) //random filename or not?
```
remove comment

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+                var fileType = Path.GetExtension(file.FileName);
+                var isNessesaryFormat = AudioExtensions.AvalableAudutionExtensions.Any(e => e.Contains(fileType.ToLower()));
+
+                if (isNessesaryFormat)
```
! and throw ValidationException

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+
+            return await _filesRepository.AddAsync(fileEntry);
+        }
+
```
remove space

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,60 @@
+        {
+            return await _filesRepository.GetAsync(fileEntry);
+        }
+        public async Task<List<FileEntry>> GetFilesAsync()
```
add space

romanstat commented on InternshipBE/ExaLearn.Dal/Database/ExaLearnDbContext.cs:
```diff
@@ -0,0 +1,14 @@
+        public ExaLearnDbContext(DbContextOptions<ExaLearnDbContext> options) : base(options)
+        {
+
+        }
```
space

romanstat commented on InternshipBE/ExaLearn.Dal/Interfaces/IGenericRepository.cs:
```diff
@@ -0,0 +1,18 @@
+{
+    public interface IGenericRepository<T> where T : class
+    {
+
```
refactoring

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/GenericRepository.cs:
```diff
@@ -0,0 +1,54 @@
+{
+    public class GenericRepository<T> : IGenericRepository<T> where T : class
+    {
+
```
remove space

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/GenericRepository.cs:
```diff
@@ -0,0 +1,54 @@
+    {
+
+        private ExaLearnDbContext appDbContext;
+        private DbSet<T> dbSet;
```
add space

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/GenericRepository.cs:
```diff
@@ -0,0 +1,54 @@
+        {
+            this.appDbContext = appDbContext;
+            dbSet = appDbContext.Set<T>();
+        }
```
add space

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/GenericRepository.cs:
```diff
@@ -0,0 +1,54 @@
+    public class GenericRepository<T> : IGenericRepository<T> where T : class
+    {
+
+        private ExaLearnDbContext appDbContext;
```
_appDbContext

romanstat commented on InternshipBE/ExaLearn.Tests/UnitTest1.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace ExaLearn.Tests
+{
+    public class UnitTest1
```
remove file

romanstat commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,35 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace WebApi.Controllers
```
ExaLearn.WebApi

romanstat commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,35 @@
+    [ApiController]
+    public class FilesController : ControllerBase
+    {
+
```
remove space

romanstat commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,35 @@
+            return Ok(await _fileService.AddAsync(file));
+        }
+
+
```
remove space

romanstat commented on InternshipBE/WebApi/Controllers/FilesController.cs:
```diff
@@ -0,0 +1,35 @@
+
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(FileEntry fileEntry)
```
FileEntry -> id
Get -> GetById

vlad12345123 committed:
Small Edits

Refactoring, rename WebApi, small edits in logic

romanstat commented on InternshipBE/ExaLearn.Bl/Extensions/AudioExtensions.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    internal static class AudioExtensions
+    {
+        public static readonly string[] AvalableAudutionExtensions =
```
Available

romanstat commented on InternshipBE/ExaLearn.Bl/Services/FileService.cs:
```diff
@@ -0,0 +1,63 @@
+{
+    public class FileService : IFileService
+    {
+        IGenericRepository<FileEntry> _filesRepository;
```
private readonly

romanstat commented on InternshipBE/WebApi/Startup.cs:
```diff
@@ -32,9 +32,12 @@ public void ConfigureServices(IServiceCollection services)
             {
                 c.SwaggerDoc("v1", new OpenApiInfo { Title = "WebApi", Version = "v1" });
             });
+            //services.AddScoped(typeof(IGenericRepository<>), typeof(GenericRepository<>));
```
remove comment

vlad12345123 committed:
refactoring

vlad12345123 committed:
refactoring

vlad12345123 committed:
refactoring

romanstat commented on InternshipBE/ExaLearn.Dal/Repositories/GenericRepository.cs:
```diff
@@ -0,0 +1,53 @@
+{
+    public class GenericRepository<T> : IGenericRepository<T> where T : class
+    {
+        private ExaLearnDbContext _appDbContext;
```
protected readonly

vlad12345123 committed:
edit modifiers access

vlad12345123 committed:
Refactoring

Change method name, change route
